src/python/pants/backend/jvm/tasks/coursier_resolve.py

```diff
@@ -160,11 +176,17 @@ def resolve(self, targets, compile_classpath, sources, javadoc):
         for conf, result_list in results.items():
           for result in result_list:
             self._load_json_result(conf, compile_classpath, coursier_cache_dir, invalidation_check,
-                                   pants_jar_base_dir, result)
+                                   pants_jar_base_dir, result, self._override_classifiers_for_conf(conf))
 
         self._populate_results_dir(vt_set_results_dir, results)
         resolve_vts.update()
 
+  def _override_classifiers_for_conf(self, conf):
+    if conf == 'src_doc': #TODO encapsulate this in the result instead of here.
```

Comment:
By this I mean that I think that Coursier's json output should note what the forced classifiers were. I think that's a more invasive change though. I could add an issue on Coursier and link it here rather than having a bare TODO.